gr-blocks/python/blocks/qa_mute.py

```diff
@@ -74,6 +75,18 @@ def test_mute_cc(self):
         op = blocks.mute_cc(True)
         self.help_cc((src_data,), expected_result, op)
 
+    def test_unmute_ff(self):
+        src_data = [1.5, 2.5, 3.5, 4.5, 5.5]
+        expected_result = [1.5, 2.5, 3.5, 4.5, 5.5]
+        op = blocks.mute_ff(False)
+        self.help_ff((src_data,), expected_result, op)
+
+    def test_mute_ff(self):
+        src_data = [1.5, 2.5, 3.5, 4.5, 5.5]
+        expected_result = [0.0, 0.0, 0.0, 0.0, 0.0]
+        op = blocks.mute_ff(True)
+        self.help_ff((src_data,), expected_result, op)
+
```

Comment:
Can you come up with a unit test that actually test the PMT handling abilities?

I know that's super annoying to do *reliably* with asynchronous message passing and synchronous stream that might be over very quickly, but that's why I extended the throttle block to be more useful for tests:

flowgraph would be something like

source (50 constants, e.g. 1.0) -> throttle (rate = 200 per second; max items per chunk: 2) -> mute -> sink

Then you `tb.start(); sleep(0.1); send_message_to_mute(); tb.wait()` and check whether in the data there's 1.0 until some point after the 20th item and from that first zero on only zeros till the end?

Reply:
I tried to add tests for the PMT's but I couldn't figure out this exact issue. Thank you, I'll add them now.

Reply:
no, thank you, your work is awesome!